### PR TITLE
Create reverseInGroupsOfK.cpp

### PR DESCRIPTION
**Pull request by RK-Shandilya**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [ ] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [x] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

**Other information:**

### CODE REVIEW

**Comment by RK-Shandilya** [comment]
// function for reversing a linkedlist in groups of size K 
// Time Complexity = O(N) and Space Complexity = O(N)

Node* kReverse(Node* head, int k) {

    // base case
    if(head == NULL) {
        return NULL;
    }

    Node* next = NULL;
    Node* curr = head;
    Node* prev = NULL;
    int count= 0;

    while( curr != NULL && count < k ) {
        next = curr -> next;
        curr -> next = prev;
        prev = curr;
        curr = next;
        count++;
    }

    if(next != NULL) {
        head -> next = kReverse(next,k);
    }
    return prev;
}

File: algorithms/CPlusPlus/Linked-Lists/reverseInGroupsOfK.cpp
Comment (aayush6200) [inline algorithms/CPlusPlus/Linked-Lists/reverseInGroupsOfK.cpp:4]: If a LinkedList is there, then the space complexity would be O(1)

Comment (RK-Shandilya) [reply algorithms/CPlusPlus/Linked-Lists/reverseInGroupsOfK.cpp:4]: No, recursion stack space is there bro.

File: algorithms/CPlusPlus/Linked-Lists/reverseInGroupsOfK.cpp
Comment (aayush6200) [inline algorithms/CPlusPlus/Linked-Lists/reverseInGroupsOfK.cpp:30]: If it is a recursive call, then call the function providing the initial arguments

**Review by aayush6200 — COMMENTED** [review]
Everything looks good

**Comment by RK-Shandilya** [comment]
thanks bro.

**Review by aayush6200 — APPROVED** [review]

**Review by aayush6200 — APPROVED** [review]

**Comment by RK-Shandilya** [comment]
Can you please tell me what I have to do now ?

**Comment by aayush6200** [comment]
after it is  approved, you can merge it

**Review by RK-Shandilya — COMMENTED** [review]
check Approve.

**Comment by RK-Shandilya** [comment]
> Everything looks good

can you please approve it ??

**Comment by RK-Shandilya** [comment]
> > Everything looks good
> 
> can you please approve it ??

??

**Review by aayush6200 — APPROVED** [review]

**Comment by RK-Shandilya** [comment]
Bro do you konw why it is showing "At least 1 approving review is required to merge this pull request" this ?? if yes then please tell me.

**Comment by RK-Shandilya** [comment]
??

**Comment by RK-Shandilya** [comment]
> Bro do you konw why it is showing "At least 1 approving review is required to merge this pull request" this ?? if yes then please tell me.

??

**Comment by RK-Shandilya** [comment]
@aayushjain7 Could you please review and approve it ??

**Comment by RK-Shandilya** [comment]
@ming-tsai Sorry sir i remove @aayushjain7 by mistake Can you please correct it ??

**Pull request closed without merging**
